### ISSUE TITLE
Unfurl urls in messages to show a handy metadata card inline

### ISSUE THREAD

**Issue by nikolaiwarner** [cabal-club/cabal-desktop#13]
When possible, show a card of information about urls found in messages after the message.

An example how slack handles this:
![image](https://user-images.githubusercontent.com/40796/41042126-dba26ece-696e-11e8-935a-1919110a3607.png)

**Comment by callil** [comment]
a few options
https://docs.microlink.io/sdk/
https://www.npmjs.com/package/react-native-link-preview

**Comment by nikolaiwarner** [comment]
Oh microlink looks great!

**Comment by callil** [comment]
I'm going to give it a shot today

**Comment by cblgh** [comment]
hm microlink requires an api-key and has rate-limiting. from what i can see [`react-native-link-preview`](https://github.com/ospfranco/react-native-link-preview/blob/master/index.js) doesn't, so let's use that?

btw i feel this feature is almost going into preferences territory and i.e. need to be toggleable (in one of the slacks i am in link previews are disabled since they take up a lost of space). is this just me or how do other people feel?

from a purely pragmatic perspective i'd prioritize issues that won't tread in preferences territory / require settings

**Comment by callil** [comment]
yes - maybe time to start on this preferences pane. Would be nice to basically relegate all _new features_ to the pane initially maybe? kind of a staging ground for things.

**Comment by nikolaiwarner** [comment]
@cblgh thanks, i agree.. we dont want a dependency like that

_usually_ i like seeing link previews but i very much understand that many would not. so we dont have to rush this. maybe we should rush the settings screen :D

there's cases where showing the a preview content of the link is quite awesome... like a youtube embed, or a soundcloud player, or a tweet ... those are the ones i really appreciate since it saves me from having to leave the chat client

**Comment by nikolaiwarner** [comment]
@callil that sounds like a fine strategy!

**Comment by callil** [comment]
I wonder if we might need to give level of priority to issues soon (p0, p1, p2 etc)
